bilby/gw/detector/interferometer.py

```diff
@@ -299,22 +300,34 @@ def get_detector_response(self, waveform_polarizations, parameters, frequencies=
         not provided, the response is computed using
         :code:`self.frequency_array`. If the frequencies are
         specified, no frequency masking is performed.
+
         Returns
         =======
         array_like: A 3x3 array representation of the detector response (signal observed in the interferometer)
+
+        Notes
+        =====
+        If the :code:`reference_time` attribute is not :code:`None`, this is
+        used to set the time at which the antenna response is evaluated,
+        otherwise the provided :code:`Parameters["geocent_time"]` is used.
         """
         if frequencies is None:
             frequencies = self.frequency_array[self.frequency_mask]
             mask = self.frequency_mask
         else:
             mask = np.ones(len(frequencies), dtype=bool)
 
+        if self.reference_time is None:
+            antenna_time = parameters["geocent_time"]
+        else:
+            antenna_time = self.reference_time
+
         signal = {}
         for mode in waveform_polarizations.keys():
             det_response = self.antenna_response(
                 parameters['ra'],
                 parameters['dec'],
-                parameters['geocent_time'],
+                antenna_time,
```

Comment:
Similarly, the input time to `self.time_delay_from_geocenter` should be replaced by `antenna_time`.